include/benchmark/benchmark.h

```diff
@@ -1203,6 +1204,24 @@ class LambdaBenchmark : public Benchmark {
 };
 #endif
 
+inline void helper_printer() {
```

Comment:
does this need to be inline? (nit, it should be HelperPrinter)

Reply:
Thanks to catch my naming convention error, sorry my last C experience changes my naming convention in my mind :/ Sorry.

In addition, I use the inline function only because I have some redefinition function at compiler time. 

The solution, in this case, can be:

- define the prototype of the function in the .h file and add the definition in the .cc file;
- use the nullprt in the function and define an HelperPrinter in the .cc file, so we can hide all the logic.

In both cases, we have some not clear states of the HelperPrintf function that is used to store the procedure to print the help information.

Maybe it is my missing some knowledge of c++ in this case?

Reply:
defining the prototype (declaring) in the header and having the implementation (definition) in the .cc file seems reasonable to me.  there's no reason i can see to have the implementation in the header.

Reply:
>defining the prototype (declaring) in the header and having the implementation (definition) in the .cc file seems reasonable to me. there's no reason i can see to have the implementation in the header.

With the following solution, I had some problems with the dynamic linker (https://github.com/vincenzopalazzo/benchmark/runs/5204928772?check_suite_focus=true) and I implemented your first suggestion to use the `NULL` pointer.

Open to other suggestions to improve the actual solution.

Reply:
it looks like you maybe had the definition in the wrong place... maybe it wasn't in the `benchmark` namespace but in the `benchmark::internal` namespace.

Reply:
ops, maybe I made this stupid error, but was a good moment to try the following solution, that introduces less code, that it is always good IMHO